bluemira/radiation_transport/advective_transport.py

```diff
@@ -322,8 +322,18 @@ def _analyse_DN(self, first_wall):
         self._make_flux_surfaces_ib()
 
         # Find the intersections of the flux surfaces with the first wall
-        for flux_surface in self.flux_surfaces:
-            flux_surface.clip(first_wall)
+        for no, flux_surface in enumerate(self.flux_surfaces):
+            try:
+                flux_surface.clip(first_wall)
+            except ValueError:
+                bluemira_warn(
+                    f"No intersection detected betwenn wall and flux_surface {no}"
+                )
+                if no > len(self.flux_surfaces_ib_lfs) and no < len(
+                    self.flux_surfaces_ob_lfs
+                ):
```

Comment:
Why the if statement ?

Reply:
In the solver, there are 4 groups of flux lines: 1) LFS_down, LFS_up, HFS_down, HFS_up.

Afterwards the flux lines from these different groups are grouped together so 
`len(flux_surfaces) = len(self.flux_surfaces_ob_lfs) + len(self.flux_surfaces_ob_hfs) + len(self.flux_surfaces_ib_lfs) + len(self.flux_surfaces_ib_hfs)`

As we know that the problem of not finding intersections happens only at the outboard, and as in this specific function we are picking flux lines from both the big pool flux_surfaces and the 4 sub-pools, I use this if statement to only check the flux lines at the outboard. 
Not sure it's clear :(

Reply:
This is dangerously dependent on the order of the flux lines additions.

I think it is better to replace this logic by comparing the fluxline themselves (if np.compare(flux_surface_ob_xxx[no]), flux_surface[no])  pop this thing out!

Reply:
Else try another logic or something ...

Reply:
Better 

Keep it like that but add a specific test !